sim/midas/src/main/cc/bridges/plusargs.cc

```diff
@@ -0,0 +1,139 @@
+#ifdef PLUSARGSBRIDGEMODULE_struct_guard
+
+#include "plusargs.h"
+#include <iomanip>
+#include <iostream>
+
+/**
+ * The default value is passed here (via the FireSim-generated.const.h).
+ * All macros should be passed to this constructor.
+ *
+ * The plusargs string is parsed out from the name. Then the argv are searched
+ * for any matches. The value of the first match is found, or else default.
+ * The value is checked to reject any runtime values that are too big for the
+ * bit_width
+ * @param [in] args The argv as a vector
+ * @param [in] mmio_addrs MMIO as provided by .h
+ * @param [in] name_orig name string of the plusargs
+ * @param [in] default_value The default value if no matching plusargs is
+ * provided
+ * @param [in] bit_width The number of bits
+ * @param [in] slice_count The number of MMIO used to represent the value
+ * @param [in] slice_addrs The MMIO addresses of the slices
+ */
+plusargs_t::plusargs_t(simif_t *sim,
+                       std::vector<std::string> &args,
+                       PLUSARGSBRIDGEMODULE_struct *mmio_addrs,
+                       const std::string name_orig,
+                       const std::string default_value,
+                       const uint32_t bit_width,
+                       const uint32_t slice_count,
+                       const uint32_t *slice_addrs)
+    : bridge_driver_t(sim), mmio_addrs(mmio_addrs), slice_count(slice_count),
+      slice_addrs(slice_addrs, slice_addrs + slice_count) {
+  std::string name = name_orig;
```

Comment:
Instead of creating a copy, please consider doing these operations on a `std::string_view`.